### PR TITLE
Show user friendly attribute names in picker

### PR DESCRIPTION
**Pull request by spacegaier**
## Proposed change

Follow-up to https://github.com/home-assistant/frontend/pull/7313.

1. <s>Fix regex for "id" to "ID" replacement by using word boundaries (to not convert "humidity" to "humIDity")</s> => already covered by https://github.com/home-assistant/frontend/pull/7448
2. Use the same conversion logic to show user friendly attribute names in the picker. The actual returned value to the card / editor / dialog is still the original real snake_case attribute name. Only the visual display part is changed. Works both in the newly changed weather card editor (https://github.com/home-assistant/frontend/pull/7335) as well as in the existing use cases such as the automation editor.
3. Fixes the attribute display, that previously broke attribute names into multiple lines due to missing `flex-grow` (see "Warning 1").
4. Allowed the attribute values to use up to half the popup size for the non-narrow display.
5. Prevent capitalization in more-info-sun (Previously showed "Setting **I**n x hours", now "Setting **i**n x hours").

image

image

**Before:**
image

**After:**
image

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [X] The code change is tested and works locally.
- [X] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/components/entity/ha-entity-attribute-picker.ts [hunk 1]
```diff
@@ -150,6 +156,8 @@ class HaEntityAttributePicker extends LitElement {
   }
 
   private _setValue(value: string) {
+    // Ensure that we send the correct internal snake_case name to the callbacks / events
+    value = formatAttributeNameIn(value);
```
Comment (bramkragten) [inline src/components/entity/ha-entity-attribute-picker.ts:160]: We should not do this, we should keep the true value and not calculate back and forth.

Comment (spacegaier) [reply src/components/entity/ha-entity-attribute-picker.ts:160]: I am not sure I understand your point. To not break / disrupt the calling card / editor / ... that uses the picker we have to provide back the internal snake_case attribute name.

Is your point coming from a performance perspective in the sense that we should ensure we only convert the snake_case values to the user friendly values once when the picker is first shown (plus if the items change) and then keep them in a class member. And then look up the matching value in a dict so we do not have to convert it back later, but just need a simple look-up instead?

Comment (bramkragten) [reply src/components/entity/ha-entity-attribute-picker.ts:160]: I have multiple problems with it, and yes one is performance. The other would be that we might get in a position that we can't simply reverse the process to get the same value back.

Can't we just add the attribute name as a property we can access on the selected event?

Comment (bramkragten) [reply src/components/entity/ha-entity-attribute-picker.ts:160]: We can use `itemLabelPath` and `itemValuePath`? <https://www.webcomponents.org/element/vaadin/vaadin-combo-box/elements/vaadin-combo-box-light#property-itemLabelPath>

Comment (spacegaier) [reply src/components/entity/ha-entity-attribute-picker.ts:160]: I am not sure if those vaadin properties will work with the input field. Getting nicer values rendered in the dropdown can also easily be done via the vaadin renderer function (I used that at the beginning). But then we still ended up with the internal name in the input field after the user had selected a value from the dropdown menu.

File: src/dialogs/more-info/controls/more-info-sun.ts [hunk 2]
```diff
@@ -72,6 +73,12 @@ class MoreInfoSun extends LitElement {
         flex-direction: row;
         justify-content: space-between;
       }
+      .time {
```
Comment (bramkragten) [inline src/dialogs/more-info/controls/more-info-sun.ts:76]: No need for a class here, you can use `ha-relative-time` as a selector

**Review by bramkragten — APPROVED** [review]

**Review by bramkragten — APPROVED** [review]

**Pull request merged**